lib/cretonne/src/ir/dfg.rs

```diff
@@ -344,14 +344,36 @@ impl DataFlowGraph {
         DisplayInst(self, inst)
     }
 
-    /// Get the value arguments on `inst` as a slice.
+    /// Get all value arguments on `inst` as a slice.
     pub fn inst_args(&self, inst: Inst) -> &[Value] {
-        self.insts[inst].arguments(&self.value_lists)
+        let (fixed, variable) = self.insts[inst].arguments(&self.value_lists);
+        if fixed.len() != 0 { fixed } else { variable }
```

Comment:
This is not correct since some instructions have both fixed and variable arguments. Indirect calls, for example.